lm-test/lm_test/cluster/cluster_interface.py

```diff
@@ -0,0 +1,111 @@
+from lm_test.config import settings
+from lm_test.utils import run
+
+
+def setup():
+    """
+    Setup the cluster where the integration test will run.
+    """
+    print("Setting up cluster...")
+
+    print("Backing up license server paths...")
+    # backup the license server paths before changing it with LM-SIM
+    license_servers_configs = [
+        "lmutil-path",
+        "rlmutil-path",
+        "lsdyna-path",
+        "lmxendutil-path",
+        "olixtool-path",
+    ]
+
+    backed_up_license_servers = {}
+
+    for license_server in license_servers_configs:
+        old_path = run(
+            "juju",
+            "config",
+            "license-manager-agent",
+            license_server,
+        )
+        backed_up_license_servers[license_server] = old_path
+        print(f"{license_server} path backed up")
+
+    print("Setting up LM-SIM...")
+    # set the lmutil path to the fake one use by LM-SIM
+    run(
+        "make",
+        "setup",
+        "-C",
+        str(settings.LM_SIM_PATH),
+        f"lm_sim_ip={settings.LM_SIM_BASE_URL}",
+    )
+    print("LM-SIM setup complete")
+
+    print("Triggering reconciliation...")
+    # run reconciliation to update license counters
+    run(
+        "juju",
+        "ssh",
+        "license-manager-agent/leader",
+        "sudo",
+        "/bin/bash",
+        "-c",
+        "'source",
+        "/srv/license-manager-agent-venv/bin/activate",
+        "&&",
+        "reconcile'",
+    )
```

Comment:
It might be good to use `shlex.split()` for these run commands so that the command is more readable:

```suggestion
    run_args = shlex.split(
        "juju ssh license-manager-agent/leader sudo /bin/bash -c "
        "'source /srv/license-manager-agent-venv/bin/activate && reconcile'"
    )
    run(*run_args)
```

Alternatively, the `run()` function could itself call `shlex.split()` and take a single string as an argument.

Reply:
For this command `shlex` ignores the single quote, making the command fail 😿

Reply:
I'll leave it like this for now.